src/producer.rs

```diff
@@ -0,0 +1,72 @@
+use crate::SharedFluvio;
+
+use log::debug;
+use fluvio::TopicProducer;
+use fluvio::FluvioError;
+
+use node_bindgen::derive::node_bindgen;
+use node_bindgen::core::NjError;
+use node_bindgen::core::val::JsEnv;
+use node_bindgen::core::TryIntoJs;
+use node_bindgen::sys::napi_value;
+use node_bindgen::core::JSClass;
+
+pub struct TopicProducerWrapper {
+    client: SharedFluvio,
+    topic: String,
+}
+
+impl TopicProducerWrapper {
+    pub fn new(client: SharedFluvio, topic: String) -> Self {
+        Self { client, topic }
+    }
+}
+
+impl TryIntoJs for TopicProducerWrapper {
+    fn try_to_js(self, js_env: &JsEnv) -> Result<napi_value, NjError> {
+        debug!("converting FluvioWrapper to js");
+        let new_instance = TopicProducerJS::new_instance(js_env, vec![])?;
+        debug!("instance created");
+        TopicProducerJS::unwrap_mut(js_env, new_instance)?.set_client(self.client);
+        TopicProducerJS::unwrap_mut(js_env, new_instance)?.set_topic(self.topic);
+        Ok(new_instance)
+    }
+}
+
+pub struct TopicProducerJS {
+    inner: Option<SharedFluvio>,
+    topic: Option<String>,
```

Comment:
Are these Options due to some limitation with node_bindgen? Or is that just how the library design worked out? If at all possible, we should try to avoid internal Options so that we don't have to double-check ourselves all throughout the API

Reply:
`inner` need to be make it optional.  but not sure about `topic1`

Reply:
Maybe we can fix this in another PR. I agree its not ideal; I need to understand the `new_instance` node_bindgen macro a bit better on how to pass in values.